### ISSUE TITLE
With large number of timers, bun consumes excessive amount of CPU time

### ISSUE THREAD

**Issue by milaz** [oven-sh/bun#4286]
### What version of Bun is running?

0.8.0+c60385716b7a7ac9f788cdf7dfe37250321e0670

### What platform is your computer?

Linux 6.4.11-060411-generic x86_64 x86_64

### What steps can reproduce the bug?

run the code:

```js
let v = 0;
for (let i = 0; i < 1e6; i++) {
	setInterval(() => v++, 5000);
}

setInterval(() => console.log(v), 5000);
```

### What is the expected behavior?

While CPU fans stay quiet as is the case with node.js, every 5 seconds numbers are printed:

1000000
2000000
3000000
4000000

### What do you see instead?

Numbers are printed correctly, but CPU usage peaks at 100%

### Additional information

Run the same code in node.js for the reference, please.

**Comment by Jarred-Sumner** [comment]
yeah we need to change how this works, the current implementation is very inefficient on linux

**Comment by milaz** [comment]
Congratulations on improving timers in Bun 1.0.19, as stated in the blog post.
However, I see the same CPU load when running the code sample from the original report.
Bun uses about 150% CPU time, but node.js barely uses 15% CPU for the same code on the same machine.

**Comment by cirospaciari** [comment]
> Congratulations on improving timers in Bun 1.0.19, as stated in the blog post.
> However, I see the same CPU load when running the code sample from the original report.
> Bun uses about 150% CPU time, but node.js barely uses 15% CPU for the same code on the same machine.

Probably related to #7790 fix already implemented in https://github.com/oven-sh/bun/pull/7792

**Comment by milaz** [comment]
Thank you. In 1.0.20 it got really better.
Instead of 150%, I see 90% CPU load from time to time.
Still, in this test, node.js outperforms Bun by a large margin.

**Comment by milaz** [comment]
Completely fixed in [v1.1.11](https://bun.sh/blog/bun-v1.1.11)
